### PR TITLE
Fixed Update School Information on Form 

### PR DESCRIPTION
**Pull request by maxsebso**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/866744728/pulses/1470811483)

## This PR

Basically uses the hidden field from the SchoolResponse and fills it into the PUT SchoolRequest for updating school info. However, in order to do this we had to temporarily make hidden not readonly because we are adding a new field to the object before submitting. 

A more permant solution is to have a different route that does not expect the hidden field in the request body OR to have the hidden field be optional in the PUT Request body.

## Screenshots
Screen Shot 2021-07-15 at 2 53 06 PM

### CODE REVIEW

File: src/containers/schoolInfo/index.tsx [hunk 1]
```diff
@@ -27,11 +31,12 @@ const SchoolInformation: React.FC = () => {
...
-  const handleFinish = (
+  const handleFinish = (schoolInfo: SchoolResponse) => (
     schoolRequest: SchoolRequest,
     editMade: boolean,
   ): void => {
     if (editMade && schoolId) {
+      schoolRequest.hidden = schoolInfo.hidden;
       dispatch(updatedSchoolRequest(schoolId, schoolRequest));
     }
     history.push('/school-contacts');
```
Comment (rymaju) [inline src/containers/schoolInfo/index.tsx:42]: Wait are we routing before we even know if the PUT request succeeded? If so then this should probably change

Comment (maxsebso) [reply src/containers/schoolInfo/index.tsx:42]: yes we are not checking currently. I will fix

**Review by rymaju — APPROVED** [review]
LGTM

**Review by granttebeau — APPROVED** [review]
LGTM

**Pull request merged**